governance/api-reviews.md

```diff
@@ -0,0 +1,207 @@
+different behavior, we strongly prefer a new version of the API over breaking
+backwards compatibility. Note that we are free to deprecate APIs; we just cannot
+break code which relies on their documented behavior. We need to worry about
+backward compatibility both of our python APIs and of the serialized GraphDefs,
+and in general breaking serialized GraphDefs is worse than breaking the python
+APIs.
+
+Forward compatibility is more subtle: we should avoid changing the graph
+produced by currently correct python code without a three weeks notice . This
+comes up most frequently when adding new ops, but also applies to non-obvious
+things such as the graph emitted by gradients or pfor.
+
+Including the name “experimental” in an API endpoint allows you to break
+backwards compatibility in the future, as noted above. However, we still prefer
+to mark the experimental API as deprecated for one release before removing it in
+the subsequent release. Please do not use the experimental namespace as an
+escape hatch for bad code or functionality you
+don’t-really-want-but-need-for-now; experimental APIs should be APIs that we
+intend to make standard in the near future, but have some lingering questions to
+resolve first.
+
+
+### Docstrings 
+
+TF APIs should have comprehensive documentation in the form of docstrings. If at
+all possible these docstrings should have runnable examples, and these examples
+should form a doctest so they stay correct. The examples should demonstrate an
+end-to-end user workflow, such that it’s clear how to generate the necessary
+inputs for the API and what to do with the outputs. The docstring should be
+understandable by someone who is not familiar with TF. See the [guide to writing
+TF docstrings](https://www.tensorflow.org/community/contribute/docs_ref) for
+more information.
+
+Our documentation generator for classes only sees methods, so prefer defining
+members as properties instead of assigning them in `__init__`.
+
+Docstrings should only refer to other public TF API symbols (i.e. do not refer
+to other symbols defined in the same file as a function which is just now being
+made public) and should refer to public API symbols by their full exported name.
+
+### Common names
+
+
+Prefer keepdims over keep_dims. Prefer axis over dim. Data types are called
+dtype. name is a common last argument of ops but backward compatibility mandates
+that new arguments are added after the last existing argument, even if that
+results in name not being the last argument.
+
+We generally prefer spelling things out over using abbreviations except when
+abbreviations are more standard than spelling things out (i.e. don’t spell out
+linalg or svd). When in doubt we ask domain experts or use web search to see
+what spelling is most common.
+
+If possible we prefer to name things in a similar way to numpy (e.g., we would
+not pick einsum as a name, but numpy and MatLab before it has, and that
```

Comment:
nit: I'm pretty sure einsum is not an official matlab function. I only see a version on MathWork's FileExchange website: https://www.mathworks.com/matlabcentral/fileexchange/68995-einsum